<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowTitle">
   <string>Firefly Control</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>images/icon.png</normaloff>images/icon.png</iconset>
  </property>
  <widget class="QWidget" name="mainWidget">
   <layout class="QGridLayout" name="mainLayout" rowstretch="1,0,0,0" columnstretch="0,1">
    <item row="0" column="0">
     <widget class="VerticalScrollArea" name="programScrollArea">
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollWidget">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <layout class="QHBoxLayout" name="programmingLayout">
        <property name="margin">
         <number>0</number>
        </property>
        <item>
         <widget class="QWidget" name="insertCommandWidget" native="true">
          <layout class="QVBoxLayout" name="insertCommandLayout" stretch="1">
           <property name="spacing">
            <number>2</number>
           </property>
           <property name="margin">
            <number>0</number>
           </property>
           <item>
            <spacer name="insertCommandSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QWidget" name="commandsWidget" native="true">
          <layout class="QVBoxLayout" name="commandsLayout" stretch="0,1">
           <property name="spacing">
            <number>2</number>
           </property>
           <property name="margin">
            <number>0</number>
           </property>
           <item>
            <widget class="QWidget" name="commandsHeaderWidget" native="true">
             <layout class="QHBoxLayout" name="programHeaderLayout">
              <item>
               <widget class="QWidget" name="selectColorHeader" native="true"/>
              </item>
              <item>
               <widget class="QLabel" name="morphHeader">
                <property name="font">
                 <font>
                  <weight>75</weight>
                  <bold>true</bold>
                 </font>
                </property>
                <property name="text">
                 <string> Fade in (ms)</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="delayHeader">
                <property name="font">
                 <font>
                  <weight>75</weight>
                  <bold>true</bold>
                 </font>
                </property>
                <property name="text">
                 <string> Stay for (ms)</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="loopHeader">
                <property name="font">
                 <font>
                  <weight>75</weight>
                  <bold>true</bold>
                 </font>
                </property>
                <property name="text">
                 <string>Loop</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <spacer name="commandsSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QWidget" name="consoleWidget" native="true">
      <layout class="QVBoxLayout" name="consoleLayout">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>1</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>2</number>
       </property>
       <item>
        <widget class="QPlainTextEdit" name="logTextEdit">
         <property name="minimumSize">
          <size>
           <width>150</width>
           <height>0</height>
          </size>
         </property>
         <property name="font">
          <font>
           <family>Courier, Courier New, monospace</family>
           <pointsize>7</pointsize>
          </font>
         </property>
         <property name="statusTip">
          <string>Terminal console: &lt; denotes input from device, &gt; output to device, other lines are program status messages and diagnostics</string>
         </property>
         <property name="lineWrapMode">
          <enum>QPlainTextEdit::NoWrap</enum>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="0">
     <layout class="QHBoxLayout" name="controlLayout">
      <item>
       <widget class="CompactButton" name="resetButton">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="statusTip">
         <string>Reset connection</string>
        </property>
        <property name="text">
         <string>Reset</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="PortLabel" name="portLabel">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>65</width>
          <height>0</height>
         </size>
        </property>
        <property name="font">
         <font>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="statusTip">
         <string>Current connected port</string>
        </property>
        <property name="styleSheet">
         <string notr="true">background: rgb(192,192,192)</string>
        </property>
        <property name="text">
         <string>No COM</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="1">
     <widget class="ConsoleEdit" name="consoleEdit">
      <property name="placeholderText">
       <string>Enter console command here</string>
      </property>
     </widget>
    </item>
    <item row="2" column="0" colspan="2">
     <widget class="QSlider" name="graphSlider">
      <property name="styleSheet">
       <string notr="true">border: 1px solid black</string>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item row="3" column="0" colspan="2">
     <widget class="QLineEdit" name="programEdit">
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <widget class="QMenu" name="fileMenu">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="quitAction"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="aboutAction"/>
   </widget>
   <addaction name="fileMenu"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="quitAction">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="statusTip">
    <string>Quit</string>
   </property>
  </action>
  <action name="aboutAction">
   <property name="text">
    <string>About</string>
   </property>
   <property name="statusTip">
    <string>About</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PortLabel</class>
   <extends>QLabel</extends>
   <header>FireflyControl</header>
  </customwidget>
  <customwidget>
   <class>ConsoleEdit</class>
   <extends>QLineEdit</extends>
   <header>FireflyControl</header>
  </customwidget>
  <customwidget>
   <class>CompactButton</class>
   <extends>QPushButton</extends>
   <header>FireflyControl</header>
  </customwidget>
  <customwidget>
   <class>VerticalScrollArea</class>
   <extends>QScrollArea</extends>
   <header>FireflyControl</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>logTextEdit</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>quitAction</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
  </connection>
 </connections>
</ui>
